src/arrayMethodJoin.js

```diff
@@ -4,9 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function (separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
+      } else {
+        result += `${this[i]}`;
```

Comment:
You're using template literals for a single variable. It's more efficient to use the variable directly without template literals. Replace `${this[i]}` with this[i].